arduino-ide-extension/src/common/protocol/ide-updater.ts

```diff
@@ -2,6 +2,13 @@ import { JsonRpcServer } from '@theia/core/lib/common/messaging/proxy-factory';
 import { Event } from '@theia/core/lib/common/event';
 import { UpdateChannel } from '../../browser/arduino-preferences';
 
+export type UpdateProgress = {
+  progressInfo?: ProgressInfo | undefined;
+  downloadFinished?: boolean;
+  downloadStarted?: boolean;
+  error?: Error;
```

Comment:
I think `error` (type of `Error`) won't go through the wire via JSON-RPC as desired. It will be an empty object (`{}`). Probably we should try.

Reply:
And anyway, this is not used in the backend, so I can move it away from `common`. Moving it.